### PR TITLE
Add a 10% stake visibility threshold on Motions page

### LINKED ISSUE #3767: Show 10% stake visibility threshold on Motions page
### Story

As a user staking on a Motion, if the staking threshold is not above 10% then the motion does not appear within the Actions list. This can be confusing in cases.
As a user staking on a Motion, I would like to be able to see what stake is required to ensure the Motion appears within the Actions list.

### Description

Add an indicator line to the Staked Progress Bar to show the 10% threshold for making a Motion visible within the Actions list. This also includes adding a changing colour state for the progress bar as an additional indicator. As well as Tooltips to provide more information.

Add a percentage of the required stake value to the Staking Slider value to show how much of the total required stake will be staked. This will help not only indicate the 10% threshold is being met during the staking process, but also help improve the clarity of the impact their stake will have. There is also a tooltip on this percentage value with additional information.

### Design

- [Figma Link - Updates to Staking Progress](https://www.figma.com/file/CCgGsBzhX3BUENS1IvgP7N/Motions-%26-Disputes?node-id=6692%3A184845)
- [Figma Link - Updates to Staking Slider](https://www.figma.com/file/CCgGsBzhX3BUENS1IvgP7N/Motions-%26-Disputes?node-id=6692%3A184845)

Screenshot 2022-08-25 101534

Screenshot 2022-08-25 101557

### PR DESCRIPTION
**Pull request by Julianahlk**
## Description

Add an indicator line to the Staked Progress Bar to show the 10% threshold for making a Motion visible within the Actions list. This also includes adding a changing color state for the progress bar as an additional indicator. As well as Tooltips to provide more information.

Add a percentage of the required stake value to the Staking Slider value to show how much of the total required stake will be staked. This will help not only indicate the 10% threshold is being met during the staking process, but also help improve the clarity of the impact their stake will have. There is also a tooltip on this percentage value with additional information.

##Design
[Figma Link - Updates to Staking Progress](https://www.figma.com/file/CCgGsBzhX3BUENS1IvgP7N/Motions-%26-Disputes?node-id=6692%3A184845)
[Figma Link - Updates to Staking Slider](https://www.figma.com/file/CCgGsBzhX3BUENS1IvgP7N/Motions-%26-Disputes?node-id=6692%3A184845)

**New stuff** ✨
Screen Shot 2022-09-15 at 12 22 09 PM

Screen Shot 2022-09-15 at 12 07 32 AM

Screen Shot 2022-09-15 at 12 22 51 PM

Screen Shot 2022-09-15 at 12 23 27 PM

Screen Shot 2022-09-15 at 12 26 01 PM

**Changes** 🏗

## TODO

- [x] Add an indicator line to the Staked Progress Bar positioned to 10% of the total size of the progress bar using css
- [x] Add a QuestionMarkIcon to the right side of the stake text 
- [x] Add a tooltip before the 10% threshold in the progress bar
- [x] Show yellow bar color below 10% threshold
- [x] Show green bar color over the 10% threshold
- [x] Show yellow percentage of amount required below 10% threshold
- [x] Show green percentage of amount required above 10% threshold

## Not Covered in this PR 

To be updated with the future UI/UX overhaul: 
* Progress bar re design 
* Staking slider re design  

Declutter section - tooltip:
* A question mark tooltip was already added for the "Select the amount to back the motion" in another PR in the Decisions branch. See [here](https://github.com/JoinColony/colonyDapp/issues/3741) if you are curious. Not sure if we needed to repeat some of the work here?

---

Resolves #3767

### CODE REVIEW

**Comment by Julianahlk** [comment]
Hey @arrenv could you please help me with clarify some questions regarding the following:

1. Are we removing completely the red banner for the "Motion requires at least 10% stake to appear in the actions list."

Screen Shot 2022-09-15 at 12 02 30 AM

And replacing it with the Stake Required tag?
Screen Shot 2022-09-15 at 12 01 55 AM

2. Is the placement of the tooltip ok if left on the left side of the progress bar for the 10% threshold?

Screen Shot 2022-09-15 at 12 07 32 AM

**Comment by arrenv** [comment]
> 1. Are we removing completely the red banner for the "Motion requires at least 10% stake to appear in the actions list."
> And replacing it with the Stake Required tag?

I had not noticed that tag change. I suppose it could be possible, but I don't think it adds too much for the effort it would take to make the change. So, I think leave the banner and the 'Staking' tag as they are. Thanks for checking.

> 2. Is the placement of the tooltip ok if left on the left side of the progress bar for the 10% threshold?

Yes, this placement is fine.

**Comment by Julianahlk** [comment]
> > 1. Are we removing completely the red banner for the "Motion requires at least 10% stake to appear in the actions list."
> >    And replacing it with the Stake Required tag?
> 
> I had not noticed that tag change. I suppose it could be possible, but I don't think it adds too much for the effort it would take to make the change. So, I think leave the banner and the 'Staking' tag as they are. Thanks for checking.
> 
> > 2. Is the placement of the tooltip ok if left on the left side of the progress bar for the 10% threshold?
> 
> Yes, this placement is fine.

Thanks so much for clarifying @arrenv 😉.

**Review by jakubcolony — COMMENTED** [review]
Looks and works great to me! 
image

image

image

File: src/modules/core/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -39,20 +41,35 @@ const ProgressBar = ({
...
         >
-          <span className={styles.thresholdPercentage}>{threshold}%</span>
+          <span
+            className={classnames(styles.thresholdPercentage, {
+              [visible]: hidePercentage,
+            })}
+          >
+            {threshold}%
+          </span>
+
```
Comment (arrenv) [inline src/modules/core/components/ProgressBar/ProgressBar.tsx:72]: Why is this needed in here?

**Review by arrenv — CHANGES_REQUESTED** [review]
Nice @Julianahlk. Just a couple of comments.

I noticed that after you stake, the "% of required" resets back to 0 - 100%. This should use the already staked as the minimum. So, in this example it would show 23% - 100%.

staking-indicator


It looks like in the design the indicator sits 1px above and below the slider where is it sits 3px above and below in this one. Also, it has a 3px radius.

Design
Screenshot 2022-09-16 102339

PR
32477710-484040aa2a1a69ebe3bfb0ab2fce9d1e

**Comment by chinins** [comment]
> I noticed that after you stake, the "% of required" resets back to 0 - 100%. This should use the already staked as the minimum. So, in this example it would show 23% - 100%.
> 

@arrenv, are you talking about the slider? In your video the behaviour seems correct (this is how the slider is working now - it shows you the % of the total stake that you would be staking). Or am I missing something?

File: src/modules/core/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (chinins) [inline src/modules/core/components/ProgressBar/ProgressBar.tsx:72]: I think that there is an unnecessary empty line here

**Review by chinins — APPROVED** [review]
Other than the indicator styles mentioned by Arren, everything looks good. Preapproving anyway as it should be an easy fix.

**Comment by arrenv** [comment]
> > I noticed that after you stake, the "% of required" resets back to 0 - 100%. This should use the already staked as the minimum. So, in this example it would show 23% - 100%.
> 
> @arrenv, are you talking about the slider? In your video the behaviour seems correct (this is how the slider is working now - it shows you the % of the total stake that you would be staking). Or am I missing something?

As discussed on Discord, I think either is fine. But agree it is a little bit clearer to be 0% - 100%.

File: src/modules/core/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (Julianahlk) [reply src/modules/core/components/ProgressBar/ProgressBar.tsx:72]: Are you referring to [visible]: hidePercentage or something else? If you are talking about the hide percentage, this is because for this progress bar we need to hide the threshold percentage but not remove it completely because it is used for the other progress bar showing the threshold percentage on top.

**Comment by Julianahlk** [comment]
> Nice @Julianahlk. Just a couple of comments.
> 
> ~I noticed that after you stake, the "% of required" resets back to 0 - 100%. This should use the already staked as the minimum. So, in this example it would show 23% - 100%.~
> 
> UPDATE: Ignore this, this is how we were thinking about it in Product. However, it is still relevant to be 0 - 100% of what is required left to stake.
> 
> staking-indicator [ staking-indicator ](https://user-images.githubusercontent.com/33682027/190594022-9dda3ea4-0093-4c5f-867c-70aa7596c588.gif) [ ](https://user-images.githubusercontent.com/33682027/190594022-9dda3ea4-0093-4c5f-867c-70aa7596c588.gif)
> 
> It looks like in the design the indicator sits 1px above and below the slider where is it sits 3px above and below in this one. Also, it has a 3px radius.
> 
> Design Screenshot 2022-09-16 102339
> 
> PR 32477710-484040aa2a1a69ebe3bfb0ab2fce9d1e

Hey @arrenv, thank you for your comments and clarifications about the threshold indicator and the staking slider over Discord. 

I made the changes you requested for the indicator. Whenever you have time please take a look. Thanks again! 
Screen Shot 2022-09-16 at 4 47 04 PM

**Review by arrenv — CHANGES_REQUESTED** [review]
@Julianahlk thank you for the update. Looks good, I think just one last change to make it clearer.

With some of the Motion already staked, the colour change of `X% of required` should factor in what is required to get over 10% of the required.

For example in my screenshot, 5% is already staked, however, the text is still yellow until it crosses 10%. However, it would actually turn green at a lower percentage i.e. 5% in this case.

32545057-175a9a10fe2238f391ddd7d2082c5470

For the green colour change to make sense, I think this would be good to account for.

**Comment by Julianahlk** [comment]
> @Julianahlk thank you for the update. Looks good, I think just one last change to make it clearer.
> 
> With some of the Motion already staked, the colour change of `X% of required` should factor in what is required to get over 10% of the required.
> 
> For example in my screenshot, 5% is already staked, however, the text is still yellow until it crosses 10%. However, it would actually turn green at a lower percentage i.e. 5% in this case.
> 
> 32545057-175a9a10fe2238f391ddd7d2082c5470
> 
> For the green colour change to make sense, I think this would be good to account for.

Hey @arrenv, I made the change based on your feedback. Let me know what you think:

staking1
staking2

**Review by arrenv — APPROVED** [review]
Working great, thank you @Julianahlk, nice work!

**Comment by Julianahlk** [comment]
> Working great, thank you @Julianahlk, nice work!

Thanks Arren!
